src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
-        /// <summary>
-        /// Writes each DLL file path along with either a namespace, class or function which contains tests(dependent on level specified by user)
-        /// </summary>
-        /// <param name="writer"> StreamWriter used to write the dll path and data, either to a txt file or stdout </param>
-        /// <param name="dll"> The dll path which is written </param>
-        /// <param name="data"> The test information corrosponding to the dll file </param>
-        private static void WriteTestDllAndName(TextWriter writer, string dll, string data)
+        public static string ScopeLevel
         {
-             writer.WriteLine("\""+dll+"\"" + " | " + data);
-             
+            get
+            {
+                return _options.Level;
+            }
+            set
+            {
+                _options.Level = value;
+            }
         }
 
         /// <summary>
-        /// Uses reflection in order to search user submitted dlls to find either namespaces, classes, or functions which contain tests, 
-        /// then uses WriteTestDllAndName() to  write data to either stdout or a txt file
+        /// Parses string into proper Level
         /// </summary>
-        /// <param name="dlls">The dlls which are submitted by user</param>
-        /// <param name="level">Specifies the level of granuality to use for the output: either namepspace, class or function</param>
-        /// <param name="writer">The StreamWriter which will be passed to WriteTestDllAndName() </param>
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
-        public static void GetTests(IList<string> dlls, Level level, TextWriter writer)
+        /// <param name="levelToParse">The string to parse</param>
+        /// <returns>The Level from parsing the string</returns>
+        public static Level ParseLevel(string levelToParse)
         {
-            List<string> testsWritten = new List<string>();
+            ThrowIf.ArgumentNull(levelToParse, nameof(levelToParse));
 
-            List<string> categoryNames = new List<string>();
+            return (Level)Enum.Parse(typeof(Level), levelToParse, ignoreCase: true);
+        }
 
-            if (dlls != null)
-            {
-                foreach (string dll in dlls)
-                {
-                    Assembly assembly = null;
-                    try
-                    {
-                        assembly = Assembly.LoadFrom(dll);
-                    }
-                    catch (FileNotFoundException e)
-                    {
-                        Console.Error.WriteLine("{0}: file {1} was not found on system", e.GetType(), dll);
-                        continue;
-                    }
+        /// <summary>
+        /// Create the output file's parent directory if it is necessary
+        /// </summary>
+        private static void CreateOutputFile()
```

Comment:
[Should] It looks like this function only creates the parent directory now, so a better name would be CreateParentDirectory()

Reply:
Fixed